internal/providers/terraform/aws/elasticache_replication_group.go

```diff
@@ -22,7 +23,19 @@ func NewElastiCacheReplicationGroup(d *schema.ResourceData, u *schema.UsageData)
 		cacheEngine = d.Get("engine").String()
 	}
 
-	if d.Get("cluster_mode").Exists() {
+	// the aws_elasticache_replication_group state output has cluster_mode set as
+	//
+	//   cluster_mode {
+	//    replicas_per_node_group = 1
+	//    num_node_groups         = 1
+	//  }
+	//
+	// even when cluster mode is disabled. This causes an issue with infracost diff as it sees no change in prices
+	// as both the new and the old resource have cluster_mode with 1 1 set. In order to circumvent this problem we
+	// need to explicitly check that cluster_enabled attribute (output attribute) is set to false in the terraform state.
+	// This will only be present in a state/diff run and won't be available in breakdown or output runs.
+	clusterDisabled := d.Get("cluster_enabled").Type != gjson.Null && d.Get("cluster_enabled").Bool()
```

Comment:
```suggestion
	clusterDisabled := d.Get("cluster_enabled").Type != gjson.Null && !d.Get("cluster_enabled").Bool()
```

Is there a golden test we could add for this?

Reply:
I guess we'd have to do a `diff` golden file test, I can look into something

Reply:
We've done them before by adding a sanitized plan JSON to the test data (e.g. https://github.com/infracost/infracost/blob/master/cmd/infracost/testdata/terraform_v0.12_plan.json).

But I don't think it's important, so I wouldn't spend time on it just now.

Reply:
good spot on the `!` i had altered this because of a `lint` error and changed the stmt